transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -342,19 +342,12 @@ public final ChannelPipeline addFirst(ChannelHandler... handlers) {
     @Override
     public final ChannelPipeline addFirst(EventExecutorGroup executor, ChannelHandler... handlers) {
         ObjectUtil.checkNotNull(handlers, "handlers");
-        if (handlers.length == 0 || handlers[0] == null) {
-            return this;
-        }
 
-        int size;
-        for (size = 1; size < handlers.length; size ++) {
-            if (handlers[size] == null) {
-                break;
-            }
-        }
-
-        for (int i = size - 1; i >= 0; i --) {
+        for (int i = handlers.length - 1; i >= 0; i --) {
             ChannelHandler h = handlers[i];
+            if (h == null) {
+                continue;
+            }
             addFirst(executor, null, h);
```

Comment:
```suggestion
            if (h != null) {
                addFirst(executor, null, h);
            }
```

Reply:
Can change. Was trying to minimize lines changed, but will clean it up.